lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Cluster.hs

```diff
@@ -207,46 +212,47 @@ withCluster config@Config{..} faucetFunds onClusterStart = runClusterM config
+                    genesisFiles
+                    cfgLastHardFork
+                    pool0port
+                    cfgNodeLogging
+                    cfgNodeOutputFile
+                    NothingK
 
-            lift $ extraClusterSetupUsingNode configuredPools runningPool0
+            let relayNodeParams =
+                    NodeParams
+                        { nodeGenesisFiles = genesisFiles
+                        , nodeHardForks = cfgLastHardFork
+                        , nodePeers = (extraPort, poolsTcpPorts)
+                        , nodeLogConfig =
+                            LogFileConfig
+                                { minSeverityTerminal = Info
+                                , extraLogDir = Nothing
+                                , minSeverityFile = Info
+                                }
+                        , nodeParamsOutputFile =
+                            cfgNodeOutputFile
+                        , nodeSocket = JustK cfgNodeToClientSocket
+                        }
+            relayNode <-
+                ContT $ withRelayNode relayNodeParams cfgRelayNodePath
+            lift $ extraClusterSetupUsingNode configuredPools relayNode
 
             case NE.nonEmpty otherPools of
-                Nothing -> liftIO $ onClusterStart runningPool0
+                Nothing -> liftIO $ onClusterStart relayNode
                 Just others -> do
-                    let relayNodeParams =
-                            NodeParams
-                                { nodeGenesisFiles = genesisFiles
-                                , nodeHardForks = cfgLastHardFork
-                                , nodePeers = (extraPort, poolsTcpPorts)
-                                , nodeLogConfig =
-                                    LogFileConfig
-                                        { minSeverityTerminal = Info
-                                        , extraLogDir = Nothing
-                                        , minSeverityFile = Info
-                                        }
-                                , nodeParamsOutputFile =
-                                    cfgNodeOutputFile
-                                }
-                    _ <-
-                        ContT
-                            $ launchPools
-                                others
-                                genesisFiles
-                                poolPorts
-                                runningPool0
-                    runningNode <-
-                        ContT $ withRelayNode relayNodeParams cfgRelayNodePath
-                    liftIO $ onClusterStart runningNode
+                    passing ()
+                        $ launchPools
+                            others
+                            genesisFiles
+                            poolPorts
+                    liftIO $ onClusterStart relayNode
   where
+    passing :: b -> (ClusterM a -> ClusterM a) -> ContT a ClusterM b
+    passing x f = ContT $ \k -> f $ k x
```

Comment:
```suggestion
```

I recommend to use plain `void` at the usage site here — by putting `f` *after* the continuation `k`, this function can do serious violence to control flow; I'd like to stick to direct use of the `ContT` constructor.